x-pack/plugin/rank-rrf/src/main/java/org/elasticsearch/xpack/rank/rrf/RRFRetrieverBuilder.java

```diff
@@ -158,6 +173,14 @@ protected RRFRankDoc[] combineInnerRetrieverResults(List<ScoreDoc[]> rankResults
         for (int rank = 0; rank < topResults.length; ++rank) {
             topResults[rank] = sortedResults[rank];
             topResults[rank].rank = rank + 1;
+            // Ensure scores are properly propagated for inner hits
```

Comment:
Is it a different bug? If so, it's better to open a separate PR to address different issues.